tests/rbd_mirror/test_rbd_snapshot_mirroring_with_upgrade.py

```diff
@@ -0,0 +1,245 @@
+                    raise_exception=True,
+                    **kw,
+                )
+
+        client.exec_command(cmd="ceph osd unset noout", sudo=True)
+        client.exec_command(cmd="ceph osd unset noscrub", sudo=True)
+        client.exec_command(cmd="ceph osd unset nodeep-scrub", sudo=True)
+    except Exception as e:
+        log.error(f"Cluster upgrade failed with error {e}")
+        return 1
+    return 0
+
+
+def run(**kw):
+    """CEPH-83574895 - Configure two-way rbd-mirror on Stand alone
+    CEPH cluster on image (with replicated and ec pools)with snapshot
+    based mirroring and perform upgrade
+    Pre-requisites :
+    We need atleast one client node with ceph-common, fio and rbd-nbd packages,
+    conf and keyring files in both clusters with snapshot based RBD mirroring
+    enabled between the clusters.
+    kw:
+        clusters:
+            ceph-rbd1:
+            config:
+                rep_pool_config:
+                num_pools: 1
+                num_images: 5
+                size: 10G
+                mode: image # compulsory argument if mirroring needs to be setup
+                mirrormode: snapshot
+                snap_schedule_levels:
+                    - image
+                snap_schedule_intervals: #one value for each level specified above
+                    - 5m
+                io_percentage: 30 #percentage of space in each image to be filled
+                ec_pool_config:
+                num_pools: 1
+                num_images: 5
+                mode: image # compulsory argument if mirroring needs to be setup
+                mirrormode: snapshot
+                snap_schedule_levels:
+                    - image
+                snap_schedule_intervals:
+                    - 5m
+                io_percentage: 30
+                command: start
+                service: upgrade
+                verify_cluster_health: true
+            ceph-rbd1:
+            config:
+                rep_pool_config:
+                num_pools: 1
+                num_images: 5
+                size: 10G
+                mode: image # compulsory argument if mirroring needs to be setup
+                mirrormode: snapshot
+                snap_schedule_levels:
+                    - image
+                snap_schedule_intervals: #one value for each level specified above
+                    - 5m
+                io_percentage: 30 #percentage of space in each image to be filled
+                ec_pool_config:
+                num_pools: 1
+                num_images: 5
+                mode: image # compulsory argument if mirroring needs to be setup
+                mirrormode: snapshot
+                snap_schedule_levels:
+                    - image
+                snap_schedule_intervals:
+                    - 5m
+                io_percentage: 30
+                command: start
+                service: upgrade
+                verify_cluster_health: true
+    Test Case Flow
```

Comment:
please add test case flow in the beginning of the module.

Reply:
i have been adding it to run.py for all the modules which i have automated, if you insist on keeping it uniform then we will have to change everywhere.